src/game/common/skirmishbattlehonors.cpp

```diff
@@ -27,3 +39,617 @@ int SkirmishBattleHonors::Get_Endurance_Medal(Utf8String name, int level) const
+        Insert_Battle_Honor(listbox, image, 0, STREAK, &row, &column, 0, bestWinStreak);
+    }
+
+    int wins = BattleHonors->Get_Wins();
+    str.Format(L"%10d", wins);
+    if (wins >= 10000) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("Domination_10000");
+        Insert_Battle_Honor(listbox, image, 1, DOMINATION, &row, &column, 0, wins);
+    } else if (wins >= 1000) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("Domination_1000");
+        Insert_Battle_Honor(listbox, image, 1, DOMINATION, &row, &column, 0, wins);
+    } else if (wins >= 500) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("Domination_500");
+        Insert_Battle_Honor(listbox, image, 1, DOMINATION, &row, &column, 0, wins);
+    } else if (wins >= 100) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("Domination_100");
+        Insert_Battle_Honor(listbox, image, 1, DOMINATION, &row, &column, 0, wins);
+    } else {
+        image = g_theMappedImageCollection->Find_Image_By_Name("Domination_100");
+        Insert_Battle_Honor(listbox, image, 0, DOMINATION, &row, &column, 0, wins);
+    }
+
+    // Ultimate: beat every map vs max number of hard opponents
+    char earned = 1;
+    for (auto it = g_theMapCache->begin(); it != g_theMapCache->end(); it++) {
+        if (it->second.m_isOfficial && it->second.m_isMultiplayer) {
+            int numBeat = BattleHonors->Get_Endurance_Medal(it->first, 4);
+            if (numBeat < it->second.m_numPlayers - 1) {
+                earned = 0;
+                break;
+            }
+        }
+    }
+    image = g_theMappedImageCollection->Find_Image_By_Name("Ultimate");
+    Insert_Battle_Honor(listbox, image, earned, ULTIMATE, &row, &column, 0, 0);
+
+    unsigned int preorder = 0;
+    Get_Unsigned_Int_From_Registry("", "Preorder", preorder);
+    if (preorder != 0) {
+        image = g_theMappedImageCollection->Find_Image_By_Name("OfficersClub");
+        Insert_Battle_Honor(listbox, image, 1, OFFICERS_CLUB, &row, &column, 0, 0);
+    }
+}
+
+void Battle_Honor_Tooltip(GameWindow *listbox, WinInstanceData *instance, unsigned int mouse)
+{
+    unsigned int honor;
+    int score;
+    int row, column;
+    Utf16String str;
+    Gadget_List_Box_Get_Entry_Based_On_XY(listbox, mouse & 0xFFFF, mouse >> 0x10, row, column);
+    if ((row == -1) || (column == -1)) {
+        str = g_theGameText->Fetch("TOOLTIP:BattleHonors", nullptr);
+        g_theMouse->Set_Cursor_Tooltip(str, -1, 0, 1.0f);
+    } else {
+        honor = (int)Gadget_List_Box_Get_Item_Data(listbox, row, column);
+        score = (int)Gadget_List_Box_Get_Item_Data(listbox, row - 1, column);
+        if (honor == 0) {
+            str = g_theGameText->Fetch("TOOLTIP:BattleHonors", nullptr);
+            g_theMouse->Set_Cursor_Tooltip(str, -1, 0, 1.0f);
+        } else if (honor != 0) {
+            if (honor & 0x8000000) { // not earned
```

Comment:
Is there perhaps an enum value missing for 1<<27 in SkirmishBattleHonorsTooltips?